api/exec/compose_stack.go

```diff
@@ -78,14 +71,15 @@ func (w *ComposeStackManager) Down(stack *portainer.Stack, endpoint *portainer.E
 		defer proxy.Close()
 	}
 
-	filePath := stackFilePath(stack)
+	filePaths := stackutils.GetStackFilePaths(stack)
 
-	_, err = w.wrapper.Down([]string{filePath}, url, stack.Name)
+	_, err = w.wrapper.Down(filePaths, url, stack.Name)
 	return err
 }
 
-func stackFilePath(stack *portainer.Stack) string {
-	return path.Join(stack.ProjectPath, stack.EntryPoint)
+// NormalizeStackName returns the passed stack name, for interface implementation only
```

Comment:
That was me who've added it in the first place and if memory serves me well, there was an issue with a libcompose, but not with the binary. Thus libcompose has a string replacement, while the binary implementation doesn't need it.
I can see now that since that, there was another issue found and the same fix applied to all stacks (https://portainer.atlassian.net/browse/EE-908). I cannot validate, but suppose it was the right call.
I'll go back and change the PR. If all stack managers now have the same behaviour, then we may as well refactor the normalisation method out of each individual stack manager.

Reply:
right :)